fv3core/fv3core/initialization/baroclinic.py

```diff
@@ -0,0 +1,522 @@
+    state.delp.data[nhalo + nx :, nhalo + ny :] = 0.0
+    state.pe.data[:] = 0.0
+    state.pt.data[:] = 1.0
+    state.ua.data[:] = 1e35
+    state.va.data[:] = 1e35
+    state.uc.data[:] = 1e30
+    state.vc.data[:] = 1e30
+    state.w.data[:] = 1.0e30
+    state.delz.data[:] = 1.0e25
+    state.phis.data[:] = 1.0e25
+    state.ps.data[:] = jablo_init.surface_pressure
+    eta = np.zeros(nz)
+    eta_v = np.zeros(nz)
+    islice, jslice, slice_3d, slice_2d = compute_slices(nx, ny)
+    # Slices with extra buffer points in the horizontal dimension
+    # to accomodate averaging over shifted calculations on the grid
+    isliceb, jsliceb, slice_3db, slice_2db = compute_slices(nx + 1, ny + 1)
+
+    # TODO: It would be neat to use <quantity>.view here rather than slices
+    # But doesn't work with all numpy operations, would need to either
+    # support some features or change the calculations
+    setup_pressure_fields(
+        eta=eta,
+        eta_v=eta_v,
+        delp=state.delp.data[slice_3d],
+        ps=state.ps.data[slice_2d],
+        pe=state.pe.data[slice_3d],
+        peln=state.peln.data[slice_3d],
+        pk=state.pk.data[slice_3d],
+        pkz=state.pkz.data[slice_3d],
+        ak=metric_terms.ak.data,
+        bk=metric_terms.bk.data,
+        ptop=metric_terms.ptop,
+        lat_agrid=metric_terms.lat_agrid.data[slice_2d],
+        adiabatic=adiabatic,
+    )
+    baroclinic_initialization(
+        eta=eta,
+        eta_v=eta_v,
+        peln=state.peln.data[slice_3db],
+        qvapor=state.qvapor.data[slice_3db],
+        delp=state.delp.data[slice_3db],
+        u=state.u.data[slice_3db],
+        v=state.v.data[slice_3db],
+        pt=state.pt.data[slice_3db],
+        phis=state.phis.data[slice_2db],
+        delz=state.delz.data[slice_3db],
+        w=state.w.data[slice_3db],
+        lon=metric_terms.lon.data[slice_2db],
+        lat=metric_terms.lat.data[slice_2db],
+        lon_agrid=metric_terms.lon_agrid.data[slice_2db],
+        lat_agrid=metric_terms.lat_agrid.data[slice_2db],
+        ee1=metric_terms.ee1.data[slice_3db],
+        ee2=metric_terms.ee2.data[slice_3db],
+        es1=metric_terms.es1.data[slice_3db],
+        ew2=metric_terms.ew2.data[slice_3db],
+        ptop=metric_terms.ptop,
+        adiabatic=adiabatic,
+        hydrostatic=hydrostatic,
+        nx=nx,
+        ny=ny,
+    )
+
+    p_var(
+        delp=state.delp.data[slice_3d],
+        delz=state.delz.data[slice_3d],
+        pt=state.pt.data[slice_3d],
+        ps=state.ps.data[slice_2d],
+        qvapor=state.qvapor.data[slice_3d],
+        pe=state.pe.data[slice_3d],
+        peln=state.peln.data[slice_3d],
+        pkz=state.pkz.data[slice_3d],
+        ptop=metric_terms.ptop,
+        moist_phys=moist_phys,
+        make_nh=(not hydrostatic),
+        hydrostatic=hydrostatic,
+    )
+    # TODO: when the dycore state is updated to only include
+    # quantities and no storages, remove the "_quantity" from phis, u and v
```

Comment:
You could remove this TODO here, when I go to do it this will be using vscode's refactor functionality so it will update everywhere at once (thanks to your use of a dataclass!). As it stands, this comment would probably remain after the refactor unless a reviewer catches it or I've thought long enough about this particular comment location while writing this message.
```suggestion
```

Reply:
ok thanks